File: documentation-examples-java/src/java/dev/typr/foundations/docs/core/QueryListenerMetrics.java
package dev.typr.foundations.docs.core;

import dev.typr.foundations.QueryEvent;
import dev.typr.foundations.QueryListener;

@SuppressWarnings("unused")
public class QueryListenerMetrics {
  // start
  QueryListener metricsListener(/* MeterRegistry registry */ ) {
    return new QueryListener() {
      @Override
      public void beforeQuery(String sql, java.util.Optional<String> name) {}

      @Override
      public void afterQuery(QueryEvent event) {
        // registry.timer("db.query",
        //     "name", event.name(),
        //     "outcome", "success"
        // ).record(event.elapsed());
      }

      @Override
      public void failedQuery(QueryEvent event) {
        // registry.timer("db.query",
        //     "name", event.name(),
        //     "outcome", "failure"
        // ).record(event.elapsed());
      }
    };
  }
  // stop
}

File: documentation-examples-kotlin/src/kotlin/dev/typr/foundationskt/docs/core/QueryListenerMetrics.kt
package dev.typr.foundationskt.docs.core

import dev.typr.foundationskt.*
import dev.typr.foundationskt.data.*

@Suppress("unused")
class QueryListenerMetrics {
    //start
    fun metricsListener(/* registry: MeterRegistry */): QueryListener =
        object : QueryListener {
            override fun beforeQuery(sql: String, name: java.util.Optional<String>) {}
            override fun afterQuery(event: QueryEvent) {
                // registry.timer("db.query",
                //     "name", event.name(),
                //     "outcome", "success"
                // ).record(event.elapsed())
            }
            override fun failedQuery(event: QueryEvent) {
                // registry.timer("db.query",
                //     "name", event.name(),
                //     "outcome", "failure"
                // ).record(event.elapsed())
            }
        }
    //stop
}
